src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -252,6 +256,32 @@ public sealed partial class TokenCache : ITokenCacheInternal
             }
         }
 
+        //This method pulls all of the acceess and refresh tokens from the cache and can therefore be very impactful on performance.
+        private void DumpCacheToLogs(AuthenticationRequestParameters requestParameters)
+        {
+            if (requestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Verbose))
+            {
+                IReadOnlyList<MsalAccessTokenCacheItem> accessTokenCacheItems = GetAllAccessTokensWithNoLocks(filterByClientId: false);
+                IReadOnlyList<MsalRefreshTokenCacheItem> refreshTokenCacheItems = GetAllRefreshTokensWithNoLocks(filterByClientId: false);
+                IList<MsalAccessTokenCacheKey> accessTokenCacheKeys = accessTokenCacheItems.Select(item => item.GetKey()).ToList();
```

Comment:
Add `Take(10)`? I think if the source is less than 10 items, it'll still work. Then just foreach through `accessTokenCacheKeys`.
```suggestion
                IList<MsalAccessTokenCacheKey> accessTokenCacheKeys = accessTokenCacheItems.Take(10).Select(item => item.GetKey()).ToList();
```

Reply:
@bgavrilMS Do you know, is ReadOnlyList faster than IList?

Reply:
I was looking for something like this. It works when there are less elements that what is specified.